app/models/course/condition/assessment.rb

```diff
@@ -0,0 +1,47 @@
+class Course::Condition::Assessment < ActiveRecord::Base
+  include ActiveSupport::NumberHelper
+
+  acts_as_condition
+  belongs_to :assessment, class_name: Course::Assessment.name, inverse_of: false
+
+  default_scope { includes(:assessment) }
+
+  def title
+    if minimum_grade_percentage
+      minimum_grade_percentage_display = number_to_percentage(minimum_grade_percentage,
+                                                              precision: 2,
+                                                              strip_insignificant_zeros: true)
+      self.class.human_attribute_name('title.title',
+                                      assessment_title: assessment.title,
+                                      minimum_grade_percentage: minimum_grade_percentage_display)
+    else
+      assessment.title
+    end
+  end
+
+  def satisfied_by?(course_user)
+    user = course_user.user
+
+    if minimum_grade_percentage
+      minimum_grade = assessment.maximum_grade * minimum_grade_percentage / 100.0
+      exists_graded_submissions_with_minimum_score_by_user?(user, minimum_grade)
+    else
+      exists_graded_submissions_by_user?(user)
+    end
+  end
+
+  private
+
+  def exists_graded_submissions_with_minimum_score_by_user?(user, minimum_grade)
+    assessment.submissions.by_user(user).
+      joins { answers }.
+      select { sum(answers.grade).as(grade) }.
+      group { course_assessment_submissions.id }.
+      exists?(['course_assessment_submissions.workflow_state = ? AND grade > ?',
```

Comment:
I found out that the `select` has no effect her. Do you know why? Here is the generated SQL
```
SELECT
		1 AS one
	FROM
		"course_assessment_submissions" INNER JOIN "course_experience_points_records"
			ON "course_experience_points_records" . "actable_id" = "course_assessment_submissions" . "id"
		AND "course_experience_points_records" . "actable_type" = $1 INNER JOIN "course_users"
			ON "course_users" . "id" = "course_experience_points_records" . "course_user_id" INNER JOIN "course_assessment_answers"
			ON "course_assessment_answers" . "submission_id" = "course_assessment_submissions" . "id"
	WHERE
		"course_assessment_submissions" . "assessment_id" = $2
		AND "course_users" . "user_id" = 1
		AND (
			course_assessment_submissions.workflow_state = 'graded'
			AND grade > 6
		)
	GROUP BY
		"course_assessment_submissions" . "id" LIMIT 1  [["actable_type", "Course::Assessment::Submission"], ["assessment_id", 4]]```